bindings/nodejs/lib/types/wallet/event.ts

```diff
@@ -1,28 +1,267 @@
-import type { IOutputResponse, ITransactionPayload } from '@iota/types';
+import type {
+    IOutputResponse,
+    ITransactionPayload,
+    ITransactionEssence,
+} from '@iota/types';
 import type { OutputData } from './output';
+import type { InclusionState } from './transaction';
+import { IInputSigningData, IRemainder } from '../client';
 
-/** Wallet event types */
-export type EventType =
-    | '*'
-    | 'ConsolidationRequired'
-    | 'LedgerAddressGeneration'
-    | 'NewOutput'
-    | 'SpentOutput'
-    | 'TransactionInclusion'
-    | 'TransactionProgress';
-
-export type NewOutputEvent = {
+export type TransactionId = string;
+
+/**
+ * All of the wallet event types.
+ */
+enum WalletEventType {
+    ConsolidationRequired = 0,
+    LedgerAddressGeneration = 1,
+    NewOutput = 2,
+    SpentOutput = 3,
+    TransactionInclusion = 4,
+    TransactionProgress = 5,
+}
+
+abstract class WalletEvent {
```

Comment:
Shouldn't this also have the `accountIndex` field or would this be a new class?

Reply:
What's printed in the example is `{ accountIndex: 0, event: { type: 0 } }` so you know from/for which account this event is emitted

Reply:
Ok but this was never done in bindings, right?

Reply:
I can't find where I would have removed this

Reply:
Yeah maybe it was never done
But what you get in the callback from listen() is something like
```TS
abstract class Event {
    private accountIndex: number;
    private event: WalletEvent;

    constructor(accountIndex: number, event: WalletEvent) {
        this.accountIndex = accountIndex;
        this.event = event;
    }

    /**
     * The account index for which the event was emtited.
     */
    getAccountIndex(): WalletEvent {
        return this.event;
    }

    /**
     * The wallet event.
     */
    getEvent(): WalletEvent {
        return this.event;
    }
}
```

Reply:
Yeah, I'm doing this

Reply:
With that it's possible to do 
```TS
        const callback = function (err: any, data: string) {
            const parsed = JSON.parse(data);
            const event = new Event(parsed.accountIndex, parsed.event)
            console.log('AccountIndex:', event.getAccountIndex());
            console.log('Event:', event.getEvent());
        };
```
and then get
```
AccountIndex: 0
Event: { type: 0 }
AccountIndex: 0
Event: { type: 5, progress: { type: 0 } }
```

Reply:
Can do even better than that

Reply:
Yeah it's ugly, please do even better :trollface:

Reply:
Let me know what you think and I'll replicate to the other binding

Reply:
I've done it to the old bindings as well